### ISSUE TITLE
/home/wade/

### ISSUE THREAD

**Issue by phoe** [wadehennessey/wcl#1]
`memory.c:13:38: fatal error: /home/wade/rtgc/allocate.h No such file or directory`

You need to fix paths all over your stuff. They all point to your local homedir.

**Comment by wadehennessey** [comment]
The currently checked in version of wcl is using a new real-time garbage collector and I haven't tried to make it easy for anyone other than me to use it yet.

I changed the latest git version of wcl to use the original gc it has always had and you shouldn't have any problems.

**Comment by phoe** [comment]
I still get this when building:
`memory.c:13:38: fatal error: /home/wade/rtgc/allocate.h: No such file or directory`
I'll fix these by hand and try compiling.

**Comment by phoe** [comment]
I get a segfault while building this.

http://pastebin.com/TkCJURsR

**Comment by wadehennessey** [comment]
You're trying to compile the pprint package, which hasn't been supported in many years. I should probably document this, remove it, or fix the problem. I'll try to take a quick look at it to see what's going on though. 

Also I noticed you compiled clx. That compiles, but hasn't worked correctly for a long time. Only the common lisp runtime and compiler are regularly recompiled and used. I could look at fixing clx and pprint if somebody had a real use for them.

**Comment by phoe** [comment]
Okay. I'm doing a regular make and don't know many of WCL's internals.. Please make a commit with the changes/fixes/removals and poke me when it happens, I'll try building again.

**Comment by wadehennessey** [comment]
Ok, I cleaned up the Make/build process and removed things that haven't been supported for a long time to reduce confusion. I left in clx even though it has a fatal bug because I might make that work again someday.

**Comment by wadehennessey** [comment]
I forgot to mention, you got a segfault because of  a problem that prevents some compiled .o files from loading correctly. They can always be statically linked just fine. This bug resulted from the port to
x86_64 and changes to how relocation information is handled in 64 bit shared object files.

**Comment by wadehennessey** [comment]
Hmm...I also forgot to ask again what you are trying to do.You seem to be interested in Common Lisp implementation details, especially when it comes to how CL can interact well with C/C++. Just curious. I have always had an interest in this, and I'm wondering if you do too.

p.s - your rtgc problems were just a distraction if you're really interested in wcl. I've built a fully concurrent, multi-threaded, real-time garbage collector, and I've used wcl as my most complex test bed to find problems in it and fix them.  I just was too lazy to shield my wcl/rtgc related checkins from the outside world, and you seem to have bumped into that.

**Comment by jpellegrini** [comment]
Hello;
I tried installing the bilary, but the bin files you have checked into git are linked against librtgc, so wcl doesn't work. I also tried to rebuild, but erbuilding needs a working wcl... :-)

**Comment by wadehennessey** [comment]
Oops, that was a mistake. The checked in binaries are using wcl's regular gc now, not rtgc. Remember that libs need to be installed in someplace like /lib64 or LD_LIBRARY_PATH needs to be set correctly for  the wcl binary to find libcl.so and libcom.so.  Let me know if you have a problem.

**Comment by jpellegrini** [comment]
Hm, are you sure they've been updated? I did a `git pull` and the binaries were indeed updated -- however, they seem to still be linked against librtgc:

```
ldd dist-bin/wcl 
librtgc.so => not found
ldd bin/wcl 
librtgc.so => not found
```

**Comment by wadehennessey** [comment]
You're right. I must have introduced a depedency on rtgc even when it's not supposed to be used. I didn't notice because I usually have a path to rtgc in my LD_LIBRARY_PATH.  I'll take another look.

**Comment by wadehennessey** [comment]
Ok, I've added a copy of librtgc.so to wcl/lib so that it can always be linked against even when not needed. Let me know if you still have problems.

**Comment by jpellegrini** [comment]
Thanks, I can run wcl now! :-)
